src/common/imagwebp.cpp

```diff
@@ -0,0 +1,231 @@
+/////////////////////////////////////////////////////////////////////////////
+// Name:        src/common/imagwebp.cpp
+// Purpose:     wxImage WebP handler
+// Author:      Hermann Höhne
+// Copyright:   (c) Hermann Höhne
+// Licence:     wxWidgets licence
+/////////////////////////////////////////////////////////////////////////////
+
+// based on code by Sylvain Bougnoux, Khral Steelforge and Markus Juergens
+// see https://forums.wxwidgets.org/viewtopic.php?t=39212
+
+// For compilers that support precompilation, includes "wx.h".
+#include "wx/wxprec.h"
+
+#if wxUSE_IMAGE && wxUSE_LIBWEBP
+
+#include "wx/imagwebp.h"
+#include "webp/demux.h"
+#include "webp/decode.h"
+#include "webp/encode.h"
+#include <vector>
+
+#ifndef WX_PRECOMP
+    #include "wx/intl.h"
+    #include "wx/log.h"
+#endif
+
+//-----------------------------------------------------------------------------
+// wxWEBPHandler
+//-----------------------------------------------------------------------------
+
+wxIMPLEMENT_DYNAMIC_CLASS(wxWEBPHandler, wxImageHandler);
+
+#if wxUSE_STREAMS
+
+#include <wx/mstream.h>
+#include <memory>
+#include <functional>
+
+typedef std::unique_ptr<WebPDemuxer, std::function<void(WebPDemuxer*)>> WebPDemuxerPtr;
+
+bool DecodeWebPDataIntoImage(wxImage *image, WebPData *webp_data, bool verbose) {
+    WebPBitstreamFeatures features;
+    VP8StatusCode status = WebPGetFeatures(webp_data->bytes, webp_data->size, &features);
+    if (status != VP8_STATUS_OK)
+    {
+        if (verbose)
+        {
+           wxLogError("WebP: GetFeatures not OK.");
+        }
+        return false;
+    }
+    image->Create(features.width, features.height, false); // this allocates memory
+    if (features.has_alpha)
+    {
+        // image has alpha channel. needs to be decoded, then re-ordered.
+        uint8_t * rgba = WebPDecodeRGBA(webp_data->bytes, webp_data->size, &features.width, &features.height);
+        if (nullptr == rgba)
```

Comment:
We generally avoid Yoda comparisons in wx, so it should be either `rgba == nullptr` or just `!rgba`, as you prefer.